### PR TITLE
Restructure Frontend Files

### PR DESCRIPTION
**Pull request by cormacjmollitor**
This commit moves all the API calls from React components into their own files. Additionally, this commit gets rid of Redux, React-Redux, and Redux-Thunk

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (cowmanjoe) [inline package-lock.json]: I think this has been accidentally added. There shouldn't be a package-lock.json in the root.

File: server/package-lock.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
 			"integrity": "sha512-a2+YeUjPkztKJu5aIF2yArYFQQp8d51wZ7DavSHjFuY1mqVgidGyzEQ41JIVNy82fXj8yPgy2vJmfIywgESW6w==",
 			"dev": true,
 			"requires": {
-				"@types/connect": "3.4.32",
-				"@types/node": "8.10.38"
+				"@types/connect": "*",
+				"@types/node": "*"
 			}
```
Comment (cowmanjoe) [inline server/package-lock.json]: Hmm, wonder why this file is different if there weren't changes to server/package.json. I suppose it's probably not an issue, these files change seemingly spontaneously sometimes.

**Review by cowmanjoe — CHANGES_REQUESTED** [review]
Awesome, lots of dead stuff removed, that's good. One suggestion is removing that package-lock in the root directory. I think the server/package-lock.json change is all good, I tested on develop and it seems that it is generating a new one now so maybe a package was updated or something like that.

File: package-lock.json (hunk 1 above)
Comment (cormacjmollitor) [reply package-lock.json]: Ok cool. I'll delete it and push the changes

File: server/package-lock.json (hunk 2 above)
Comment (cormacjmollitor) [reply server/package-lock.json]: I'll revert the changes on this one and push

**Review by cowmanjoe — APPROVED** [review]

**Pull request merged**